### PR TITLE
CHAT-2839: audio_device: support capture of output device audio (i.e. record desktop/games)

### PR DESCRIPTION
**Pull request by dscherba**

### CODE REVIEW

**Comment by dscherba** [comment]
//cc @jstys92 @DillonSkeehan

File: webrtc/modules/audio_device/win/audio_device_core_win.cc [hunk 1]
```diff
@@ -1980,9 +1981,10 @@ int16_t AudioDeviceWindowsCore::RecordingDevices()
 
     CriticalSectionScoped lock(&_critSect);
 
-    if (_RefreshDeviceList(eCapture) != -1)
+    if (_RefreshDeviceList(eCapture) != 1 &&
```
Comment (jstys92) [inline webrtc/modules/audio_device/win/audio_device_core_win.cc]: Const compared against changes from -1 to 1, doesn't look right

File: webrtc/modules/audio_device/win/audio_device_core_win.h [hunk 2]
```diff
@@ -238,6 +240,9 @@ class AudioDeviceWindowsCore : public AudioDeviceGeneric
     void _Lock() { _critSect.Enter(); };
     void _UnLock() { _critSect.Leave(); };
 
+private:    // Loopback capture support
```
Comment (jstys92) [inline webrtc/modules/audio_device/win/audio_device_core_win.h:243]: `private:` marked twice here

**Review by jstys92 — COMMENTED** [review]
One functional change that seems off, otherwise LGTM

File: webrtc/modules/audio_device/win/audio_device_core_win.h (hunk 2 above)
Comment (dscherba) [reply webrtc/modules/audio_device/win/audio_device_core_win.h:243]: @jstys92 : this is sort of the style of this file (having multiple `private` blocks with grouped functions)...

Comment (jstys92) [reply webrtc/modules/audio_device/win/audio_device_core_win.h:243]: Gotcha, that's fine then

File: webrtc/modules/audio_device/win/audio_device_core_win.cc (hunk 1 above)
Comment (dscherba) [reply webrtc/modules/audio_device/win/audio_device_core_win.cc]: ack, good catch

**Comment by dscherba** [comment]
Updated with fix for `-1` versus `1` thing. Tested using webrtc functional audio (unit) test -- still seems to work, which makes sense since the difference is in case of failure.

**Review by jstys92 — APPROVED** [review]
👍

**Comment by dscherba** [comment]
Moving ahead this this...

**Pull request merged**